### ISSUE TITLE
OOM in tests since Quarkus 3.13.0 (Part 2)

### ISSUE THREAD

**Issue by mschorsch** [quarkusio/quarkus#42355]
### Describe the bug

As described in https://github.com/quarkusio/quarkus/issues/42303 we get an OOM in our tests.

Because https://github.com/quarkusio/quarkus/issues/42303 doesn't fix the problem completly i've opened a new issue with a new reproducer. 

Compared to the reproducer in  https://github.com/quarkusio/quarkus/issues/42303 i have added more dependencies.

### How to Reproduce?

[oom-reproducer2.zip](https://github.com/user-attachments/files/16522904/oom-reproducer2.zip)

`./gradlew test --console=plain`

⚠️ The reproducer uses 999-SNAPSHOT instead of 3.13.1 because 3.13.1 is not available in Maven Central yet.

### Output of `java -version`

Java 21

### Quarkus version or git rev

Quarkus 3.13.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.8

**Comment by geoand** [comment]
I'll take a look

**Comment by geoand** [comment]
I think I know what the problem is, but it will take a look while before I am able to test my theory

**Comment by geoand** [comment]
Would you be able to test #42369?


For whatever reason I could not get Gradle to use my changes...

**Comment by mschorsch** [comment]
I've tested https://github.com/quarkusio/quarkus/pull/42369 against our real application and still get an OOM. 

Tomorrow i can try the reproducer.

**Comment by geoand** [comment]
Thanks for checking.

Looking forward to hearing if the OOM still happens with the reproducer

**Comment by geoand** [comment]
I was able to reproduce the problem in the reproducer even with my fix - so obviously there is even more to it.

**Comment by geoand** [comment]
I closed https://github.com/quarkusio/quarkus/pull/42369 because it does not fix the issue. 

I was however able to figure out to pinpoint the problem to the JVM and System related binders. Those use JMX which does not play nicely at all with multiple classloaders. 
One option here would be to disable these by default and have users explicitly opt-in to them. 
@ebullient WDYT?

**Comment by geoand** [comment]
@mschorsch my cmment above means that you can get the reproducer working by setting:

```properties
quarkus.micrometer.binder.jvm=false
quarkus.micrometer.binder.system=false
```

**Comment by gsmet** [comment]
Ah, that's funny, I actually identified them as problematic here: https://github.com/quarkusio/quarkus/issues/41233

Is it what you're seeing?

I really think we should try to tackle this issue and make this an error so that we can catch further issues.

**Comment by geoand** [comment]
> I really think we should try to tackle this issue

What do you mean exactly, I don't follow

**Comment by geoand** [comment]
Turns out that this can be fixed and rather easily...

**Comment by mschorsch** [comment]
> @mschorsch my comment above means that you can get the reproducer working by setting:
> 
> ```ini
> quarkus.micrometer.binder.jvm=false
> quarkus.micrometer.binder.system=false
> ```

I can confirm that this does indeed prevent the OOM in the reproducer 👍 .

In our real application, however, we still get an OOM even though I have deactivated the micrometer extension 😬 .
```yaml
quarkus:
  micrometer:
    enabled: false
    binder:
      jvm: false
      system: false
```

Even the complete removal of `io.quarkus:quarkus-micrometer-registry-prometheus` did not help.

I will test https://github.com/quarkusio/quarkus/pull/42388 but I suspect that there are other issues that are not covered by the repeoducer.

**Comment by geoand** [comment]
> I will test https://github.com/quarkusio/quarkus/pull/42388 but I suspect that there are other issues that are not covered by the repeoducer.

If you do find more issues, please open new issues with the relevent reproducers.
Thanks!

**Comment by mschorsch** [comment]
I can confirm that https://github.com/quarkusio/quarkus/pull/42388 fixes the issue the reproducer.

In our real application we still get an OOM, seems a part 3 is needed...

Thanks for your work @geoand @gsmet

**Comment by geoand** [comment]
🙏🏼

**Comment by ebullient** [comment]
> I closed #42369 because it does not fix the issue.
> 
> I was however able to figure out to pinpoint the problem to the JVM and System related binders. Those use JMX which does not play nicely at all with multiple classloaders. One option here would be to disable these by default and have users explicitly opt-in to them. @ebullient WDYT?
> 
> cc @gsmet who was also interested in the outcome of this.

I know you're chasing other things, but I think not having this by default would be a surprise.

**Comment by geoand** [comment]
In any case, that proposal is no longer necessary :)
